File: frontend/src/types/track.ts
export interface TrackLesson {
  id: string;
  title: string;
  order: number;
  completed: boolean;
}

export interface TrackModule {
  id: string;
  title: string;
  order: number;
  lessons: TrackLesson[];
  completed_count: number;
  total_count: number;
  percent_complete: number;
}

export interface TrackView {
  modules: TrackModule[];
  total_lessons: number;
  completed_lessons: number;
  percent_complete: number;
  resume_lesson_id: string | null;
}

File: backend/app/schemas/track.py
"""Schemas for the literacy track view."""
from pydantic import BaseModel


class TrackLesson(BaseModel):
    id: str
    title: str
    order: int
    completed: bool


class TrackModule(BaseModel):
    id: str
    title: str
    order: int
    lessons: list[TrackLesson]
    completed_count: int
    total_count: int
    percent_complete: float


class TrackView(BaseModel):
    modules: list[TrackModule]
    total_lessons: int
    completed_lessons: int
    percent_complete: float
    resume_lesson_id: str | None
